Sources/App/Commands/Ingest.swift

```diff
@@ -124,13 +124,30 @@ func ingest(client: Client,
                         Current.logger().warning("storeS3Readme failed")
                         s3Readme = .error("\(error)")
                     }
+                    
+                    var fork: Fork?
+                    do {
+                        if let parentUrl = metadata.repository?.normalizedParentUrl {
+                            if let packageId = try await Package.query(on: database)
+                                .filter(\.$url == parentUrl)
+                                .first()?.id {
+                                fork = .parentId(packageId)
+                            } else {
+                                fork = .parentURL(parentUrl)
+                            }
+                        }
+                    } catch {
+                        Current.logger().warning("updating forked from failed")
+                    }
```

Comment:
I'll take a proper look at this soon, but I noticed this as a first small issue. I wonder if it might be better to do this inside `updateRepository`. It would save us passing one more variable into that function as we already pass in `metadata`, which drives this logic.

I can give it a proper review tomorrow, but I thought I'd mention this first as it may change the structure of the PR a little bit.

Reply:
Thanks for the feedback. It definitely makes sense to move this to the `updateRepository`. Will do that.